From a26e4c1d5dfee603b9b861be905c799dddddc170 Mon Sep 17 00:00:00 2001
From: Prakash Brahmajyosyula <bprakash@marvell.com>
Date: Mon, 3 Dec 2018 11:34:50 +0300
Subject: [PATCH 0842/1051] soc: octeontx2: fix to generate cgx stats in
 debugfs.

With commit d20ef14b1654f34f7c83eba35932c05f35247016 ("soc: octeontx2: Fix
build warnings") removed strsep function, which is used to generate lmacid.
Now use strrchr to get lmacid from sysfs file name.

Change-Id: I3eb7e00509b2a7c697a963763a3ada02fa8d1038
Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1750
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_debugfs.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_debugfs.c b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
index 5660d472f896..6f8c06b11c93 100644
--- a/drivers/soc/marvell/octeontx2/rvu_debugfs.c
+++ b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
@@ -658,25 +658,20 @@ static ssize_t rvu_dbg_cgx_stat_display(struct file *filp,
 {
 	void *data = filp->private_data;
 	struct dentry *current_dir;
-	int err = 0, lmac_id = 0;
+	int err, lmac_id;
 	char *buf;
 
 	current_dir = filp->f_path.dentry->d_parent;
-	buf = kzalloc(strlen(current_dir->d_name.name), GFP_KERNEL);
+	buf = strrchr(current_dir->d_name.name, 'c');
 	if (!buf)
 		return count;
 
-	memcpy(buf, current_dir->d_name.name,
-	       strlen(current_dir->d_name.name));
-
-	if (kstrtoint(buf, 10, &lmac_id) >= 0) {
+	err = kstrtoint(buf + 1, 10, &lmac_id);
+	if (err >= 0) {
 		err = cgx_print_stats(data, lmac_id);
 		if (err)
-			goto stat_display_done;
+			return err;
 	}
-
-stat_display_done:
-	kfree(buf);
 	return err;
 }
 RVU_DEBUG_FOPS(cgx_stat, cgx_stat_display, NULL);
-- 
2.17.1

